Add py-atlannot version 0.1.2

Stannislav opened the pull request:
* The new version (`0.1.2`) is hosted on GitHub / PyPI (open-sourced recently)
* The old version (`0.1.1`) is still pointing to GitLab because there won't be a release for it on PyPI.

Commit:
Add py-atlannot version 0.1.2

matz-e commented on var/spack/repos/builtin/packages/py-atlannot/package.py:
```diff
@@ -10,29 +10,36 @@
-    # depends_on('py-dvc', type=('run'))
-    depends_on('py-matplotlib', type=('run'))
-    depends_on('py-numpy', type=('run'))
+    depends_on("py-antspyx@0.2.4", type=("run"))
```
All dependencies here and below should actually be `"build", "run"`

Stannislav replied:
Is that something new? Weren't all normal python dependencies usually just listed with `type=("run")`?

matz-e replied:
Didn't pay much attention to it before (and used both normally), but Gianluca pointed out at some point: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

> These packages are required for building and installation. You can add them with type=('build', 'run').

Kind of a safeguard against `pip` downloading anything required during installation with Spack.

Stannislav replied:
Fair enough, thanks for the reference.

Changed in 1c493704d7

Commit:
Add the "build" type to python dependencies

Reference:
https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

Commit:
Add https to the homepage link
